Client/Application/AppState.swift

```diff
@@ -0,0 +1,50 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+
+enum AppState {
+    case Browser(currentURL: NSURL, isBookmarked: Bool, isDesktopSite: Bool, hasAccount: Bool, isPrivate: Bool)
+    case HomePanels(selectedPanelIndex: Int, isPrivate: Bool)
```

Comment:
Using an `enum` for state seems to be stretching the meaning of an enumerated type.

Also: can we move away from calling a tab a browser.